### ISSUE TITLE
Hardcoded credentials rule will flag any parser code (like for "token")

### ISSUE THREAD

**Issue by willfaught** [GoASTScanner/gas#108]
See: https://github.com/GoASTScanner/gas/blob/master/rules/hardcoded_credentials.go#L77

This is flagged as bad:

```go
const (
	ATNStateInvalidType    = 0
	ATNStateBasic          = 1
	ATNStateRuleStart      = 2
	ATNStateBlockStart     = 3
	ATNStatePlusBlockStart = 4
	ATNStateStarBlockStart = 5
	ATNStateTokenStart     = 6
	// ...

	ATNStateInvalidStateNumber = -1
)
```

This would also be flagged as bad:

```go
const (
	UsernameFieldLabel = "Username"
	// ...
)
```

**Comment by gcmurphy** [comment]
Just merged PR #109 that will ensure the first example does not report an error. 

The second example shouldn't trigger as we only look for variables containing phrases like password, key, token or secret so 'UsernameFieldLabel' shouldn't trigger it, however the error reporting for GenDecl blocks such as that didn't reference the exact constant declaration that was causing the error, instead it was referencing the entire `const ( )` block. So I fixed that in PR #110. 

The test is still pretty noisy which is why it has been marked as a low confidence result. We are looking at introducing and additional entropy check  (see https://github.com/GoASTScanner/gas/issues/105) which should reduce these kinds of false positives.

**Comment by willfaught** [comment]
Wouldn't PasswordFieldLabel be flagged?

**Comment by gcmurphy** [comment]
Yes it would currently.  There is the option to use the flag `-exclude=G101` to exclude this rule.

**Comment by gcmurphy** [comment]
Closing. 

There isn't really a heuristic that we can use here to reduce false positives in variable naming. This test should really target things like API keys so when #105 is merged it will eliminate a large number of false positives around innocuous things like `const PasswordFieldLabel = "Enter password"`. 

In addition the option is available to exclude this test altogether via the `-exclude=G101` command line option or add a `// #nosec` comment to the affected field. (e.g. `const PasswordFieldLabel = "Enter password" // #nosec`).

**Comment by willfaught** [comment]
OK, thanks.
